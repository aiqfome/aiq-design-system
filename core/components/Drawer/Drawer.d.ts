import React, { ReactNode } from 'react';
import { DefaultTheme } from 'styled-components';
import { LayoutProps, MarginProps, ShadowProps, PaddingProps } from 'styled-system';
export interface Props extends DefaultTheme, LayoutProps, ShadowProps, MarginProps, PaddingProps {
    opened: boolean;
    loading?: boolean;
    children?: ReactNode;
    onClose?: () => void;
    variation?: 'right' | 'left';
}
export declare const DrawerStyled: import("styled-components").StyledComponent<"div", any, Props, never>;
export declare const Drawer: React.FC<Props>;
//# sourceMappingURL=Drawer.d.ts.map