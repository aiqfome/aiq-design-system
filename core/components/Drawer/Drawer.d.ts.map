{"version":3,"file":"Drawer.d.ts","sourceRoot":"","sources":["../../../src/components/Drawer/Drawer.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,EAAE,EAAE,SAAS,EAAuB,MAAM,OAAO,CAAA;AAE7D,OAAe,EAAO,YAAY,EAAE,MAAM,mBAAmB,CAAA;AAC7D,OAAO,EAKL,WAAW,EACX,WAAW,EACX,WAAW,EACX,YAAY,EACb,MAAM,eAAe,CAAA;AAKtB,MAAM,WAAW,KACf,SAAQ,YAAY,EAClB,WAAW,EACX,WAAW,EACX,WAAW,EACX,YAAY;IACd,MAAM,EAAE,OAAO,CAAA;IACf,OAAO,CAAC,EAAE,OAAO,CAAA;IACjB,QAAQ,CAAC,EAAE,SAAS,CAAA;IACpB,OAAO,CAAC,EAAE,MAAM,IAAI,CAAA;IACpB,SAAS,CAAC,EAAE,OAAO,GAAG,MAAM,CAAA;CAC7B;AAgDD,eAAO,MAAM,YAAY,uEAiBxB,CAAA;AAED,eAAO,MAAM,MAAM,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAyDlC,CAAA"}