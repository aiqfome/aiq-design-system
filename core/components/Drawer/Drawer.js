"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Drawer = exports.DrawerStyled = void 0;
const react_1 = __importStar(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importStar(require("styled-components"));
const styled_system_1 = require("styled-system");
const Flex_1 = require("../Flex");
const Loading_1 = require("../Loading");
const drawerVariations = {
    right: styled_components_1.css `
    right: 0;

    ${({ opened }) => opened &&
        styled_components_1.css `
        transform: translateX(0);
      `}

    ${({ opened }) => !opened &&
        styled_components_1.css `
        display: none;
        transform: translateX(100%);
      `}
  `,
    left: styled_components_1.css `
    left: 0;

    ${({ opened }) => opened &&
        styled_components_1.css `
        transform: translateX(0);
      `}

    ${({ opened }) => !opened &&
        styled_components_1.css `
        display: none;
        transform: translateX(-100%);
      `}
  `
};
const Mask = styled_components_1.default.div `
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1000;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.45);
`;
exports.DrawerStyled = styled_components_1.default.div `
  max-width: 760px;
  ${styled_system_1.layout}
  ${styled_system_1.shadow}
  background-color: ${props => props.theme.colors.white};
  border: 1px solid ${props => props.theme.colors.mediumGrey};
  position: fixed;
  top: 0;
  height: 100%;
  z-index: 2000;
  transition: transform 0.8s;
  ${styled_system_1.layout}
  ${styled_system_1.shadow}
  ${styled_system_1.margin}
  ${styled_system_1.padding}

  ${({ variation }) => drawerVariations[variation || 'right']}
`;
exports.Drawer = (_a) => {
    var { loading = false, onClose, opened = false, variation = 'right', children } = _a, props = __rest(_a, ["loading", "onClose", "opened", "variation", "children"]);
    const [bodyOverflow, setBodyOverflow] = react_1.useState(false);
    react_1.useEffect(() => {
        if (opened) {
            if (document.body.style.overflow !== 'hidden') {
                document.body.style.overflow = 'hidden';
                setBodyOverflow(true);
            }
        }
        else {
            if (bodyOverflow)
                document.body.style.overflow = 'auto';
        }
        return () => {
            if (bodyOverflow) {
                document.body.style.overflow = 'auto';
            }
        };
    }, [opened, bodyOverflow]);
    return (react_1.default.createElement(react_1.default.Fragment, null,
        onClose && opened && (react_1.default.createElement(Mask, { "data-testid": 'drawer-mask', onClick: onClose })),
        react_1.default.createElement(exports.DrawerStyled, Object.assign({ opened: opened, variation: variation, "data-testid": 'drawer-content', className: opened ? 'drawer-open' : 'drawer-close' }, props),
            react_1.default.createElement(Flex_1.Flex, { overflow: 'auto', height: '100%', flexDirection: 'column' }, loading ? (react_1.default.createElement(Flex_1.Flex, { flex: 1, height: '100%', alignItems: 'center', justifyContent: 'center' },
                react_1.default.createElement(Loading_1.Loading, null))) : (children)))));
};
exports.Drawer.propTypes = {
    loading: prop_types_1.default.bool,
    onClose: prop_types_1.default.func,
    opened: prop_types_1.default.bool.isRequired,
    variation: prop_types_1.default.oneOf(['right', 'left']),
    children: prop_types_1.default.node
};
