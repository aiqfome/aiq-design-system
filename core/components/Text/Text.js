"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Text = void 0;
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importStar(require("styled-components"));
const styled_system_1 = require("styled-system");
exports.Text = styled_components_1.default.span `
  ${styled_system_1.color}
  ${styled_system_1.space}
  ${styled_system_1.layout}
  ${styled_system_1.fontSize}
  ${styled_system_1.fontWeight}
  ${styled_system_1.typography}

  white-space: ${props => props.whiteSpace || 'normal'};

  ${({ cursor }) => cursor &&
    styled_components_1.css `
      cursor: ${cursor};
    `};

  ${({ truncate }) => truncate &&
    styled_components_1.css `
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    `};
`;
exports.Text.propTypes = {
    cursor: prop_types_1.default.string,
    whiteSpace: prop_types_1.default.oneOf(['nowrap', 'normal', 'pre'])
};
