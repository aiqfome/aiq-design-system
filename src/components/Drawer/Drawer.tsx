import React, { ReactNode } from 'react'
import PropTypes from 'prop-types'
import styled, { css, DefaultTheme } from 'styled-components'
import { layout, LayoutProps, shadow, ShadowProps } from 'styled-system'

export interface Props extends DefaultTheme, LayoutProps, ShadowProps {
  variation?: 'right' | 'left'
  opened: boolean
  children?: ReactNode
}

const drawerVariations: { [index: string]: any } = {
  right: css<Props>`
    right: 0;

    ${({ opened }) =>
      opened &&
      css`
        transform: translateX(0);
      `}

    ${({ opened }) =>
      !opened &&
      css`
        transform: translateX(100%);
      `}
  `,
  left: css<Props>`
    left: 0;

    ${({ opened }) =>
      opened &&
      css`
        transform: translateX(0);
      `}

    ${({ opened }) =>
      !opened &&
      css`
        transform: translateX(-100%);
      `}
  `
}

export const DrawerStyled = styled.div<Props>`  
  ${layout}
  ${shadow}
  background-color: ${props => props.theme.colors.white};
  border: 1px solid ${props => props.theme.colors.mediumGrey};
  position: fixed;
  top: 0;
  overflow: auto;
  height: 100vh;
  z-index: 2000;
  transition: transform 0.8s;

  ${({ variation }) => drawerVariations[variation || 'right']}
`

export const Drawer: React.FC<Props> = ({
  opened = false,
  variation = 'right',
  children,
  ...props
}) => {
  return (
    <DrawerStyled opened={opened} variation={variation} {...props}>
      {children}
    </DrawerStyled>
  )
}

Drawer.propTypes = {
  opened: PropTypes.bool.isRequired,
  variation: PropTypes.oneOf(['right', 'left']),
  children: PropTypes.node
}
