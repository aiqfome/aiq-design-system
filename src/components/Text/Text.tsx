import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import {
  color,
  ColorProps,
  space,
  SpaceProps,
  layout,
  LayoutProps,
  fontSize,
  FontSizeProps,
  fontWeight,
  FontWeightProps,
  typography,
  TypographyProps
} from 'styled-system'

export interface Props
  extends ColorProps,
    SpaceProps,
    LayoutProps,
    FontSizeProps,
    FontWeightProps,
    TypographyProps {
  cursor?: string
  whiteSpace?: 'nowrap' | 'normal' | 'pre'
}

export const Text = styled.span<Props>`
  ${color}
  ${space}
  ${layout}
  ${fontSize}
  ${fontWeight}
  ${typography}

  ${({ cursor }) =>
    cursor &&
    css`
      cursor: ${cursor};
    `};
  white-space: ${props => props.whiteSpace || 'normal'};
`

Text.propTypes = {
  cursor: PropTypes.string,
  whiteSpace: PropTypes.oneOf(['nowrap', 'normal', 'pre'])
}
