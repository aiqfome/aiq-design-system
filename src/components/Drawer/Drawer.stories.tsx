import React from 'react'
import { withKnobs, text, select, boolean } from '@storybook/addon-knobs'

import { Text } from '../Text'
import { Drawer } from './Drawer'

export default {
  component: Drawer,
  title: 'Drawer',
  decorators: [withKnobs]
}

export const basic: React.FC = () => (
  <Drawer
    opened={boolean('opened', true)}
    variation={select('variation', { Right: 'right', Left: 'left' }, 'right')}
  >
    <Text>{text('text', 'Design System')}</Text>
  </Drawer>
)

export const left: React.FC = () => (
  <Drawer opened={true} variation='left'>
    <Text>Design System</Text>
  </Drawer>
)

export const right: React.FC = () => (
  <Drawer opened={true} variation='right'>
    <Text>Design System</Text>
  </Drawer>
)
