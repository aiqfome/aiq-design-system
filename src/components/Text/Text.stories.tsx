import React from 'react'
import { withKnobs, text, number } from '@storybook/addon-knobs'

import { Text } from './Text'
import { Flex } from '../Flex'

export default {
  component: Text,
  title: 'Text',
  decorators: [withKnobs]
}

export const basic: React.FC = () => (
  <Text
    color={text('color', 'primary')}
    fontSize={number('fontSize', 16)}
    cursor={text('cursor', '')}
    fontWeight={text('fontWeight', 'medium')}
  >
    {text('text', 'Aiqfome')}
  </Text>
)

export const cursor: React.FC = () => (
  <Text cursor='pointer'>Design System</Text>
)

export const WhiteSpace: React.FC = () => (
  <Flex width='50px' border='1px solid black'>
    <Text whiteSpace='nowrap'>Design System</Text>
  </Flex>
)
