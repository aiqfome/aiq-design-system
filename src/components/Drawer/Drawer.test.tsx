import React from 'react'

import { Drawer } from './Drawer'
import { Text } from '../Text'

import { render } from '../utils/test/render'

describe('must match with the previous snapshot', () => {
  test('snapshot renders', () => {
    const component = render(
      <Drawer opened={true}>
        <Text>Drawer component!</Text>
      </Drawer>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
