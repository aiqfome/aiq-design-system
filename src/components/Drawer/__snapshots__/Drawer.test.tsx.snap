// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<div
  className="sc-bdnylx dTrXQw"
>
  <span
    className="sc-gtssRu kUMpil"
  >
    Drawer component!
  </span>
</div>
`;
